cmd/erasure-multipart.go

```diff
@@ -309,20 +309,21 @@ func (er erasureObjects) ListMultipartUploads(ctx context.Context, bucket, objec
 		if populatedUploadIds.Contains(uploadID) {
 			continue
 		}
-		fi, err := disk.ReadVersion(ctx, minioMetaMultipartBucket, pathJoin(er.getUploadIDDir(bucket, object, uploadID)), "", false)
-		if err != nil {
-			if !IsErrIgnored(err, errFileNotFound, errDiskNotFound) {
-				logger.LogIf(ctx, err)
+		// If present, use time stored in ID.
+		startTime := time.Now()
+		if split := strings.Split(uploadID, "x"); len(split) == 2 {
+			t, err := strconv.ParseInt(split[1], 10, 64)
+			if err == nil {
+				startTime = time.Unix(0, t)
 			}
-			// Ignore this invalid upload-id since we are listing here
-			continue
 		}
-		populatedUploadIds.Add(uploadID)
```

Comment:
In passing: Was the removal of `populatedUploadIds.Add` intentional? If so, noting is adding to `populatedUploadIds` and it looks like the check `populatedUploadIds.Contains` in line 309 can get removed.

Reply:
Let me check.

Reply:
I can't really see how ListDir can return duplicate entries. But let's re-add it, since it is a cheap sanity check.

Reply:
https://github.com/minio/minio/pull/18339